apps/s_server.c

```diff
@@ -2747,12 +2747,14 @@ static int init_ssl_connection(SSL *con)
                 wbio = SSL_get_wbio(con);
                 if (wbio) {
                     BIO_get_fd(wbio, &fd);
-                }
-
-                if (!wbio || BIO_connect(fd, client, 0) == 0) {
-                    BIO_printf(bio_err, "ERROR - unable to connect\n");
-                    BIO_ADDR_free(client);
-                    return 0;
+                } else {
+                    if (BIO_connect(fd, client, 0) == 0) {
+                        BIO_printf(bio_err, "ERROR - unable to connect\n");
+                        BIO_ADDR_free(client);
+                        return 0;
+                    }
+                    wbio = SSL_get_wbio(con);
+                    BIO_ctrl(wbio, BIO_CTRL_DGRAM_SET_CONNECTED, 0, client);
```

Comment:
This looks like it's going to fail. We know that `SSL_get_wbio` is going to return `NULL` (nothing has changed in `con` since it last returned `NULL` with this call), so the `BIO_ctrl` call will just immediately fail.

Reply:
done.